### PR TITLE
scnlib: update fast_float, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **scnlib/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: recipes/scnlib/all/conanfile.py [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 class ScnlibConan(ConanFile):
     name = "scnlib"
     description = "scanf for modern C++"
     license = "Apache-2.0"
-    topics = ("scnlib", "parsing", "io", "scanf")
-    homepage = "https://github.com/eliaskosunen/scnlib"
     url = "https://github.com/conan-io/conan-center-index"
-
+    homepage = "https://github.com/eliaskosunen/scnlib"
+    topics = ("scnlib", "parsing", "io", "scanf")
```
Comment (prince-chrismc) [inline recipes/scnlib/all/conanfile.py:18]:
```suggestion
    topics = ("parsing", "io", "scanf")
```

File: recipes/scnlib/all/conanfile.py [hunk 2]
```diff
@@ -25,98 +28,88 @@ class ScnlibConan(ConanFile):
...
-    @property
-    def _compilers_minimum_version(self):
-        return {
-            "gcc": "5",
-            "clang": "6.0",
-            "Visual Studio": "16" if tools.Version(self.version) >= "1.0" else "15",
-        }
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self, src_folder="src")
```
Comment (prince-chrismc) [inline recipes/scnlib/all/conanfile.py:50]:
```suggestion
        if self.options.header_only:
            basic_layout(self, src_folder="src")
        else:
            cmake_layout(self, src_folder="src")
```

File: recipes/scnlib/all/conanfile.py [hunk 3]
```diff
@@ -1,18 +1,21 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
-import os
+from conan import ConanFile
+from conan.tools.microsoft import check_min_vs
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
```
Comment (prince-chrismc) [inline recipes/scnlib/all/conanfile.py:6]:
```suggestion
from conan.tools.scm import Version
from conan.tools.layout import basic_layout
from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
```

**Review by prince-chrismc — COMMENTED** [review]
Should use the basic layout for header only :)

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**